mne/viz/_figure.py

```diff
@@ -0,0 +1,430 @@
+# -*- coding: utf-8 -*-
+"""Figure classes for MNE-Python's 2D plots."""
+
+# Authors: Daniel McCloy <dan@mccloy.info>
+#
+# License: Simplified BSD
+
+from matplotlib.figure import Figure
+from ..utils import set_config
+
+
+class MNEFigParams:
+    """Container for MNE figure parameters."""
+    def __init__(self, **kwargs):
+        # default key to close window
+        self.close_key = 'escape'
+        for key, val in kwargs.items():
+            setattr(self, key, val)
+
+
+class MNEFigure(Figure):
+    """Wrapper of matplotlib.figure.Figure; adds MNE-Python figure params."""
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # add our param object
+        self.mne = MNEFigParams(fig_size_px=self._get_size_px())
+
+    def _get_dpi_ratio(self):
+        """Get DPI ratio (to handle hi-DPI screens)."""
+        dpi_ratio = 1.
+        for key in ('_dpi_ratio', '_device_scale'):
+            dpi_ratio = getattr(self.canvas, key, dpi_ratio)
+        return dpi_ratio
+
+    def _get_size_px(self):
+        """Get figure size in pixels."""
+        dpi_ratio = self._get_dpi_ratio()
+        size = self.get_size_inches() * self.dpi / dpi_ratio
+        return size
+
+    def _inch_to_rel(self, dim_inches, horiz=True):
+        """Convert inches to figure-relative distances."""
+        fig_w, fig_h = self.get_size_inches()
+        w_or_h = fig_w if horiz else fig_h
+        return dim_inches / w_or_h
+
+    def _onclick_help(self, event):
+        pass
+
+
+class MNEBrowseFigure(MNEFigure):
+    """Interactive figure with scrollbars, for data browsing."""
+    def __init__(self, inst, *args, xlabel='Time (s)', show_scrollbars=True,
+                 **kwargs):
+        from matplotlib.widgets import Button
+        from mpl_toolkits.axes_grid1.axes_size import Fixed
+        from mpl_toolkits.axes_grid1.axes_divider import make_axes_locatable
+        from mne.viz.utils import _get_figsize_from_config
+
+        # figsize is the first arg of Figure()
+        figsize = args.pop(0) if len(args) else kwargs.pop('figsize', None)
+        if figsize is None:
+            figsize = _get_figsize_from_config()
+        kwargs['figsize'] = figsize
```

Comment:
```suggestion
        kwargs['figsize'] = figsize or _get_figsize_from_config()
```

is also always an option